packages/runtime/src/runtime/messaging/subservices/collector.ts

```diff
@@ -0,0 +1,47 @@
+import { uuid } from '@botpress/messaging-client'
+import { IO, Logger } from 'botpress/sdk'
+import LRUCache from 'lru-cache'
+import ms from 'ms'
+import { EventEngine } from '../../events'
+import { MessagingInteractor } from './interactor'
+import { MessagingLifetime } from './lifetime'
+
+export class MessagingCollector {
+  private collectingCache: LRUCache<string, uuid>
+
+  constructor(
+    private logger: Logger,
+    private eventEngine: EventEngine,
+    private interactor: MessagingInteractor,
+    private lifetime: MessagingLifetime
+  ) {
+    this.collectingCache = new LRUCache<string, uuid>({ max: 5000, maxAge: ms('5m') })
+  }
+
+  set(eventId: string, messageId: uuid) {
+    this.collectingCache.set(eventId, messageId)
+  }
+
+  get(eventId: string) {
+    return this.collectingCache.get(eventId)
+  }
```

Comment:
Note : this was `return this.collectingCache[eventId]` which is a bug, needs to be fixed on master also